impl/pkg/pkarr/record.go

```diff
@@ -0,0 +1,94 @@
+package pkarr
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/TBD54566975/ssi-sdk/util"
+	"github.com/anacrolix/dht/v2/bep44"
+	"github.com/anacrolix/torrent/bencode"
+)
+
+type Record struct {
+	Value          []byte   `json:"v" validate:"required"`
+	Key            [32]byte `json:"k" validate:"required"`
+	Signature      [64]byte `json:"sig" validate:"required"`
+	SequenceNumber int64    `json:"seq" validate:"required"`
+}
+
+type Response struct {
+	V   []byte   `validate:"required"`
+	Seq int64    `validate:"required"`
+	Sig [64]byte `validate:"required"`
+}
+
+func NewRecord(k []byte, v []byte, sig []byte, seq int64) (*Record, error) {
+	record := Record{SequenceNumber: seq}
```

Comment:
might be better to refactor this method to do the validations up front and then construct the object in the return